### ISSUE TITLE
Convert Sklearn Pipeline with Custom Steps

### ISSUE THREAD

**Issue by embicks** [onnx/sklearn-onnx#137]
I have written my own vectorizer class as part of an Sklearn Pipeline that I am trying to convert to ONNX.  In my convert I get the error "Unable to find a shape calculator for type '<class '__main__.myVector'>'." 

Is there a way to convert Pipelines with custom steps?

**Comment by prabhat00155** [comment]
Are you interested in contributing this to sklearn-onnx or just want to create a custom transformer for your scenario. For the former, you need to implement an operator converter under skl2onnx/operator_converter/ and a shape calculator which calculates the output type and shape, for example: https://github.com/onnx/sklearn-onnx/blob/master/skl2onnx/shape_calculators/NearestNeighbours.py
You'll also need to add these to _supported_converter.py
For the latter, there is a pending PR which would make writing custom converters easy: https://github.com/onnx/sklearn-onnx/pull/128

**Comment by xadupre** [comment]
This scenario is supported. The Following example shows one exemple: https://github.com/onnx/sklearn-onnx/blob/master/docs/examples/plot_custom_model.py. PR #128 aims at make it much more simple : https://github.com/xadupre/sklearn-onnx/blob/oo/docs/examples/plot_convert_syntax.py.

**Comment by embicks** [comment]
Thanks, 

I have followed your example and added two methods to my class like you do in https://github.com/xadupre/sklearn-onnx/blob/oo/docs/examples/plot_convert_syntax.py.

My class I am trying to incorporate has a fit method that sets a dictionary of document frequencies and a list for a vocabulary. I have defined:

    def to_onnx_operator(self, inputs=None, outputs=('Y', )):
        if inputs is None:
            raise RuntimeError("inputs should contain one name")
        i0 = self.get_inputs(inputs, 0)
        W = self.Idfs
        S = self.vocabulary
        return OnnxDiv(OnnxSub(i0, W), S,
                       output_names=outputs)

where W is the dictionary and S is the list. When I try to convert this I get the error:
TypeError: Unable to interpret the input name for type <class 'dict'>.
